modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -682,4 +683,26 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
     });
   });
 
+  it('TINY-9222: Scroll to the cursor after undo and redo', () => {
+    const editor = hook.editor();
+    editor.undoManager.clear();
+    editor.focus();
+
+    const height = 5000;
+    editor.resetContent(`<p class="first">top paragraph</p><p style="height: ${height}px"></p><p class="last">last paragraph</p>`);
+    TinySelections.select(editor, 'p.last', [ 0 ]);
+    TinyContentActions.type(editor, 'updated ');
+
+    const doc = TinyDom.document(editor);
+    const editorHeight = editor.getWin().innerHeight;
+
+    const checkScroll = (action: 'undo' | 'redo') => {
+      Scroll.to(0, 0, doc);
+      editor.undoManager[action]();
+      Scroll.intoViewIfNeeded(UiFinder.findIn(doc, 'p.last').getOrDie(), SugarElement.fromDom(doc.dom.scrollingElement as Element));
```

Comment:
It looks like  `Scroll.intoViewIfNeeded` call here is redundant since this should be done during undo operation. Try to remove it. The test should pass without it.